library/test/src/term/terminfo/mod.rs

```diff
@@ -119,13 +130,24 @@ pub(crate) struct TerminfoTerminal<T> {
 impl<T: Write + Send> Terminal for TerminfoTerminal<T> {
     fn fg(&mut self, color: color::Color) -> io::Result<bool> {
         let color = self.dim_if_necessary(color);
+        if cfg!(miri) && color < 8 {
+            // The Miri logic for this only works for the most basic 8 colors, which we just assume
+            // the terminal will support. (`num_colors` is always 0 in Miri, so higher colors will
+            // just fail. But libtest doesn't use any higher colors anyway.)
+            let s = format!("\x1B[3{color}m");
```

Comment:
Huh, slightly surprised this is more efficient under miri than something like `write!(self.out, "\x1B[3{color}m")`.

Reply:
I don't think it is more efficient. But Miri isn't *that* slow, a bit of formatting is fine.

The terminfo parser is doing a *lot* of work and constructing a huge data structure to represent the parsed result, that's why this is noticeable.

But your code looks nicer anyway. :D  So, happy to change it accordingly.